src/dstack/_internal/core/backends/runpod/compute.py

```diff
@@ -80,7 +91,25 @@ def run_job(
             ports="10022/tcp",
             bid_per_gpu=instance_offer.price if instance_offer.instance.resources.spot else None,
         )
+
         instance_id = resp["id"]
+
+        if container_registry_auth_id is not None:
+            instance_id = self.api_client.edit_pod(
```

Comment:
Please leave a comment on why this "edit" approach is implemented and possibly TODO to replace it once runpod updates the API.

Reply:
You are right, it's better to leave a comment. I have added it.